### ISSUE TITLE
Consistent Crash When Opening a Product 

### ISSUE THREAD

**Issue by shiki** [woocommerce/woocommerce-android#2777]
**Describe the bug**

App crashes when opening a product. I think it depends on the type of product because other products can be opened. 

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.woocommerce.android, PID: 21129
    java.lang.IllegalStateException: Cannot call this method while RecyclerView is computing a layout or scrolling androidx.recyclerview.widget.RecyclerView{98e0352 VFE..V... ......ID 0,0-720,0 #7f0a0103 app:id/cardsRecyclerView}, adapter:com.woocommerce.android.ui.products.adapters.ProductPropertyCardsAdapter@9e69623, layout:androidx.recyclerview.widget.LinearLayoutManager@afec720, context:com.woocommerce.android.ui.main.MainActivity@cf4d64
        at androidx.recyclerview.widget.RecyclerView.assertNotInLayoutOrScroll(RecyclerView.java:3051)
        at androidx.recyclerview.widget.RecyclerView$RecyclerViewDataObserver.onItemRangeChanged(RecyclerView.java:5547)
        at androidx.recyclerview.widget.RecyclerView$AdapterDataObservable.notifyItemRangeChanged(RecyclerView.java:12268)
        at androidx.recyclerview.widget.RecyclerView$Adapter.notifyItemRangeChanged(RecyclerView.java:7447)
        at androidx.recyclerview.widget.AdapterListUpdateCallback.onChanged(AdapterListUpdateCallback.java:60)
        at androidx.recyclerview.widget.BatchingListUpdateCallback.dispatchLastEvent(BatchingListUpdateCallback.java:67)
        at androidx.recyclerview.widget.DiffUtil$DiffResult.dispatchUpdatesTo(DiffUtil.java:852)
        at androidx.recyclerview.widget.DiffUtil$DiffResult.dispatchUpdatesTo(DiffUtil.java:802)
        at com.woocommerce.android.ui.products.adapters.ProductPropertyCardsAdapter.update(ProductPropertyCardsAdapter.kt:20)
        at com.woocommerce.android.ui.products.ProductDetailFragment.showProductCards(ProductDetailFragment.kt:282)
        at com.woocommerce.android.ui.products.ProductDetailFragment.access$showProductCards(ProductDetailFragment.kt:45)
        at com.woocommerce.android.ui.products.ProductDetailFragment$setupObservers$2.onChanged(ProductDetailFragment.kt:161)
        at com.woocommerce.android.ui.products.ProductDetailFragment$setupObservers$2.onChanged(ProductDetailFragment.kt:45)
        at androidx.lifecycle.LiveData.considerNotify(LiveData.java:131)
        at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:149)
        at androidx.lifecycle.LiveData.setValue(LiveData.java:307)
        at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
        at com.woocommerce.android.ui.products.ProductDetailViewModel.updateCards(ProductDetailViewModel.kt:523)
        at com.woocommerce.android.ui.products.ProductDetailViewModel.access$updateCards(ProductDetailViewModel.kt:86)
        at com.woocommerce.android.ui.products.ProductDetailViewModel$productDetailViewStateData$1.invoke(ProductDetailViewModel.kt:116)
        at com.woocommerce.android.ui.products.ProductDetailViewModel$productDetailViewStateData$1.invoke(ProductDetailViewModel.kt:86)
        at com.woocommerce.android.viewmodel.LiveDataDelegate$observe$1.onChanged(LiveDataDelegate.kt:43)
        at com.woocommerce.android.viewmodel.LiveDataDelegate$observe$1.onChanged(LiveDataDelegate.kt:22)
        at androidx.lifecycle.LiveData.considerNotify(LiveData.java:131)
        at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:149)
        at androidx.lifecycle.LiveData.setValue(LiveData.java:307)
        at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
        at androidx.lifecycle.SavedStateHandle$SavingStateLiveData.setValue(SavedStateHandle.java:306)
        at com.woocommerce.android.viewmodel.LiveDataDelegate.setValue(LiveDataDelegate.kt:58)
        at com.woocommerce.android.ui.products.ProductDetailViewModel.setViewState(Unknown Source:10)
        at com.woocommerce.android.ui.products.ProductDetailViewModel.updateProductDraft(ProductDetailViewModel.kt:508)
        at com.woocommerce.android.ui.products.ProductDetailViewModel.updateProductDraft$default(ProductDetailViewModel.kt:459)
        at com.woocommerce.android.ui.products.ProductDetailViewModel.onProductTitleChanged(ProductDetailViewModel.kt:407)
        at com.woocommerce.android.ui.products.ProductDetailCardBuilder$title$1.invoke(ProductDetailCardBuilder.kt:536)
        at com.woocommerce.android.ui.products.ProductDetailCardBuilder$title$1.invoke(ProductDetailCardBuilder.kt:50)
E/AndroidRuntime:     at com.woocommerce.android.ui.products.viewholders.EditableViewHolder$bind$1$1.invoke(EditableViewHolder.kt:18)
        at com.woocommerce.android.ui.products.viewholders.EditableViewHolder$bind$1$1.invoke(EditableViewHolder.kt:8)
        at com.woocommerce.android.ui.products.WCProductPropertyEditableView$setOnTextChangedListener$$inlined$doAfterTextChanged$1.afterTextChanged(TextView.kt:100)
        at android.widget.TextView.sendAfterTextChanged(TextView.java:9380)
        at android.widget.TextView.setText(TextView.java:5403)
        at android.widget.TextView.setText(TextView.java:5250)
        at android.widget.EditText.setText(EditText.java:113)
        at android.widget.TextView.setText(TextView.java:5207)
        at com.woocommerce.android.ui.products.WCProductPropertyEditableView.show(WCProductPropertyEditableView.kt:27)
        at com.woocommerce.android.ui.products.viewholders.EditableViewHolder.bind(EditableViewHolder.kt:21)
        at com.woocommerce.android.ui.products.adapters.ProductPropertiesAdapter.onBindViewHolder(ProductPropertiesAdapter.kt:78)
        at com.woocommerce.android.ui.products.adapters.ProductPropertiesAdapter.onBindViewHolder(ProductPropertiesAdapter.kt:38)
        at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7065)
        at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7107)
        at androidx.recyclerview.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:6012)
        at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6279)
        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6118)
        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6114)
        at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2303)
        at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1627)
        at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1587)
        at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:665)
        at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4134)
        at androidx.recyclerview.widget.RecyclerView.onMeasure(RecyclerView.java:3540)
        at android.view.View.measure(View.java:22071)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6602)
        at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1514)
        at android.widget.LinearLayout.measureVertical(LinearLayout.java:806)
        at android.widget.LinearLayout.onMeasure(LinearLayout.java:685)
        at android.view.View.measure(View.java:22071)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6602)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at androidx.cardview.widget.CardView.onMeasure(CardView.java:260)
        at com.google.android.material.card.MaterialCardView.onMeasure(MaterialCardView.java:159)
        at android.view.View.measure(View.java:22071)
        at androidx.recyclerview.widget.RecyclerView$LayoutManager.measureChildWithMargins(RecyclerView.java:9384)
        at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1653)
        at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1587)
        at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:665)
        at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4134)
        at androidx.recyclerview.widget.RecyclerView.onMeasure(RecyclerView.java:3540)
        at android.view.View.measure(View.java:22071)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6602)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at android.view.View.measure(View.java:22071)
        at androidx.core.widget.NestedScrollView.measureChildWithMargins(NestedScrollView.java:1599)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
E/AndroidRuntime:     at androidx.core.widget.NestedScrollView.onMeasure(NestedScrollView.java:585)
        at android.view.View.measure(View.java:22071)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6602)
        at androidx.coordinatorlayout.widget.CoordinatorLayout.onMeasureChild(CoordinatorLayout.java:760)
        at com.google.android.material.appbar.HeaderScrollingViewBehavior.onMeasureChild(HeaderScrollingViewBehavior.java:99)
        at com.google.android.material.appbar.AppBarLayout$ScrollingViewBehavior.onMeasureChild(AppBarLayout.java:1892)
        at androidx.coordinatorlayout.widget.CoordinatorLayout.onMeasure(CoordinatorLayout.java:831)
        at android.view.View.measure(View.java:22071)
        at androidx.constraintlayout.widget.ConstraintLayout.internalMeasureChildren(ConstraintLayout.java:1227)
        at androidx.constraintlayout.widget.ConstraintLayout.onMeasure(ConstraintLayout.java:1572)
        at android.view.View.measure(View.java:22071)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6602)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at android.view.View.measure(View.java:22071)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6602)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at android.view.View.measure(View.java:22071)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6602)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at android.view.View.measure(View.java:22071)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6602)
        at androidx.coordinatorlayout.widget.CoordinatorLayout.onMeasureChild(CoordinatorLayout.java:760)
        at com.google.android.material.appbar.HeaderScrollingViewBehavior.onMeasureChild(HeaderScrollingViewBehavior.java:99)
        at com.google.android.material.appbar.AppBarLayout$ScrollingViewBehavior.onMeasureChild(AppBarLayout.java:1892)
        at androidx.coordinatorlayout.widget.CoordinatorLayout.onMeasure(CoordinatorLayout.java:831)
        at android.view.View.measure(View.java:22071)
        at androidx.constraintlayout.widget.ConstraintLayout.onMeasure(ConstraintLayout.java:1676)
        at android.view.View.measure(View.java:22071)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6602)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at androidx.appcompat.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:143)
        at android.view.View.measure(View.java:22071)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6602)
        at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1514)
        at android.widget.LinearLayout.measureVertical(LinearLayout.java:806)
        at android.widget.LinearLayout.onMeasure(LinearLayout.java:685)
        at android.view.View.measure(View.java:22071)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6602)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at android.view.View.measure(View.java:22071)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6602)
        at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1514)
        at android.widget.LinearLayout.measureVertical(LinearLayout.java:806)
        at android.widget.LinearLayout.onMeasure(LinearLayout.java:685)
        at android.view.View.measure(View.java:22071)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6602)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at com.android.internal.policy.DecorView.onMeasure(DecorView.java:724)
        at android.view.View.measure(View.java:22071)
        at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:2422)
        at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1504)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1761)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1392)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6752)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
        at android.view.Choreographer.doCallbacks(Choreographer.java:723)
E/AndroidRuntime:     at android.view.Choreographer.doFrame(Choreographer.java:658)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
        at android.os.Handler.handleCallback(Handler.java:790)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
D/io.sentry.android.event.helper.AndroidEventBuilderHelper: Proguard UUIDs file not found.
```

**To Reproduce**

1. Use recent `develop` (957b7dd6c)
2. Navigate to Products
3. Open a product 

I can share credentials for my site if this makes it easier. 

**Screenshots**

Here is [a video](https://drive.google.com/file/d/15xouPHpu7Hw8tuDpsIXgpcmrZfbYowNM/view?usp=sharing).

**Expected behavior**

No crash. 🙂 

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: Android Emulator Galaxy Nexus 720p 
- Android version: 8.1 (API 27 Oreo)
- WooCommerce Android version: 

**WordPress Environment**
<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin.
```
</details>

**Comment by AmandaRiu** [comment]
Try adding an ampersand to the product title to reproduce this bug (thanks @rachelmcr !)

**Comment by anitaa1990** [comment]
Crash happened in the 4.9 beta release to 2 users, events: 9

**Comment by anitaa1990** [comment]
Just to add some notes to ☝️ , when I add `&` to the product title from `wp-admin`, the app is working fine and is not crashing. But when adding the same `&` from the app and updating the product title is causing the app to crash.

**Comment by AmandaRiu** [comment]
Crash fixed in #2788 - closing.
